### ISSUE TITLE
Feature: PickRandomSubSet to get a random subset from a list/enumerable

### ISSUE THREAD

**Issue by joelharkes** [bchavez/Bogus#49]
PickRandom is great but when i want to pick a random subset of items it gets quickly complex.

I have to PickRandom multiple times and then do a distinct to be sure to not have doubles.

It would be great if a function for this existed.

**Comment by joelharkes** [comment]
Here i found a great and simple algorithm to do it better: http://stackoverflow.com/a/48089/1275832

**Comment by joelharkes** [comment]
I implemented a possible solution based on the link shared above: 

```c#
public static class BogusExtensionMethods
{
    public static IEnumerable<T> PickRandomSubset<T>(this Faker faker, IEnumerable<T> set, int amountToTake)
    {
        var size = set.Count();
        foreach (var item in set)
        {
            if(amountToTake <= 0)
                yield break;
            if (faker.Random.Int(1, size) <= amountToTake)
            {
                amountToTake--;
                yield return item;
            }
            size--;
        }
    }
}
```

**Comment by bchavez** [comment]
Hi @joelharkes ,

Thanks for the suggestion. I'll have to let this API settle in my mind a little bit. Also, I'd like to give others a chance to voice their opinions on the usefulness of this method. I think it could be useful.

And Happy New Year.

-Brian

:watch: :city_sunset: [***"I just can't wait... I just can't wait... for saturday night..."***](https://www.youtube.com/watch?v=wOjGmpmvGz8#t=2m)

**Comment by bchavez** [comment]
Alright, we can add this, but, maybe we could rename `PickRandomSubset` to just `PickRandom` as an overload for `IEnumerable<T>`. What do you think?

Feel free to send a PR if you'd like to get this in `Bogus` faster.

Thanks,
Brian

:boom: :fire: ***["Set it ablaze like a candle wick... Light it up, light it up..."](https://www.youtube.com/watch?v=qDcFryDXQ7U)***

**Comment by joelharkes** [comment]
as you wish: https://github.com/bchavez/Bogus/pull/51

**Comment by bchavez** [comment]
Closing. Implemented as of **Bogus** `v8.0,4`. Thanks again for your PR @joelharkes.
